flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/util/HiveTableUtil.java

```diff
@@ -133,4 +140,34 @@ public static Partition createHivePartition(String dbName, String tableName, Lis
 		return partition;
 	}
 
+	public static CatalogTable toHiveCatalogTable(CatalogTable oldTable) {
```

Comment:
We may need to add comments to explain the method.

Reply:
+1, I think we should just convert the old table's TableSchema to a new TableSchema by changing the corresponding field types. This does more than what we need.

Reply:
You're right. But `CatalogTableImpl` currently does not have `setSchema` method. Should I add one?

Reply:
Probably another way is to create a converted table schema and return it in `getTableSchema` method?

Reply:
I think the current modification is the most proper modification under the current code base. The table schema in the catalog table is not only used in table source, but will also be passed into hive input format will be used there. So I think the most proper way is to modify the table schema here.

Reply:
I think this method can be renamed to something like `convertTableSchemaForHive`?